runtime/src/utils/buffer.rs

```diff
@@ -111,11 +114,12 @@ impl<B: Blob> Buffer<B> {
         let bytes_to_read = std::cmp::min(self.buffer_size as u64, blob_remaining) as usize;
 
         // Read the data - we only need a single read operation since we know exactly how much data is available
-        self.blob
-            .read_at(&mut self.buffer[..bytes_to_read], self.blob_position)
+        let buffer = std::mem::take(&mut self.buffer);
```

Comment:
I think we could do:
```
{ buffer: Option<Vec<u8>> }

let buffer  = self.buffer.take();
let result = self.blob.read_at(buffer);
self.buffer = if let Ok(b) { Some(b) } else { Some(Vec::new()) };
```

Reply:
I removed the `take` so I don't think this is necessary anymore

Reply:
see https://github.com/commonwarexyz/monorepo/pull/933#discussion_r2091252259